grc/core/generator/flow_graph.tmpl

```diff
@@ -378,6 +403,37 @@ def main(top_block_cls=$(class_name), options=None):
         #end for
     tb.Wait()
         #end if
+    #elif $generate_options == 'bokeh_gui'
+    serverProc = bokehgui.utils.create_server()
+    def killProc(signum, frame, tb):
+        tb.stop()
+        tb.wait()
+        serverProc.terminate()
+        serverProc.kill()
+    time.sleep(1)
+    try:
+        \# Define the document instance
+        doc = curdoc()
+        #if $flow_graph.get_option('author')
+        doc.title = "$title - $flow_graph.get_option('author')"
+        #else
+        doc.title = "$title"
+        #end if
+        session = push_session(doc, session_id="$flow_graph.get_option('id')",
+                               url = "http://localhost:5006/bokehgui")
+        \# Create Top Block instance
+        tb = top_block_cls(doc)
+        try:
+            tb.start()
+            signal.signal(signal.SIGTERM, functools.partial(killProc, tb=tb))
+            session.loop_until_closed()
+        finally:
+            print "Exiting the simulation. Stopping Bokeh Server"
+            tb.stop()
+            tb.wait()
+    finally:
+        serverProc.terminate()
+        serverProc.kill()
```

Comment:
Is it possible to query if
- a server instance is already running (before creating our own - possibly failing due to the tcp port being used already)
- this is the only document being served - if not the server instance should probably live on...